packages/contracts-periphery/contracts/universal/op-nft/CitizenshipChecker.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { SocialContract } from "./SocialContract.sol";
+
+interface ICitizenshipChecker {
+    function isCitizen(address _who, bytes memory _proof) external view returns (bool);
+}
+
+contract CitizenshipChecker is ICitizenshipChecker {
+    address public root;
```

Comment:
Both of these values should be immutables, they are never altered once set and are set in the constructor